plugin/fulltext/mysql-test/ngram/ngram_basic.test

```diff
@@ -0,0 +1,25 @@
+# Ngram fulltext parser basic functionality.
+
+SET NAMES utf8mb4;
+SELECT @@ngram_token_size;
+
+let $engine_index= 3;
+while ($engine_index) {
+  if ($engine_index == 3) { let $engine= InnoDB; }
+  if ($engine_index == 2) { let $engine= MyISAM; }
+  if ($engine_index == 1) { let $engine= Maria; }
+
+  let $charset_index= 6;
+  while ($charset_index) {
+    if ($charset_index == 6) { let $charset= utf8mb4; }
+    if ($charset_index == 5) { let $charset= big5; }
+    if ($charset_index == 4) { let $charset= gbk; }
+    if ($charset_index == 3) { let $charset= gb2312; }
+    if ($charset_index == 2) { let $charset= sjis; }
+    if ($charset_index == 1) { let $charset= ujis; }
```

Comment:
Could we please also test UTF-16 and UCS2? (Results may be funny, but it should not crash.)

I would strongly suggest to use a `.combinations` file to cover different character encodings. In that way, it should actually be possible to compare that all storage engines are behaving in the same way. The `.result` file should be recoded using `utf8mb4`, because the other encodings should not be a superset of that when it comes to this test case.

Reply:
> Could we please also test UTF-16 and UCS2? 

There is `ngram_ucs2.test` that tries to create fulltext indexes for columns encoded in UCS2, UTF16, and UTF32. As index creation fails, there is no much sense in adding these encodings to the `ngram_basic.test`.

> I would strongly suggest to use a `.combinations` file to cover different character encodings.

Not really possible, since results differ for different encodings. For example, big5 (with extensions) supports traditional Chinese characters, and Japanese syllables, but doesn't support simplified Chinese characters and Korean syllables. If you inspect `ngram_basic.result`, you may notice that there are question marks at the places where unsupported characters were. Same with other encodings. gb2312 cares about simplified Chinese, so it will garble Japanese which tends to use traditional characters.

UTF-8 represents all of used text just fine, but other encodings miss some parts. Supported sets differ a bit, so query results differ a bit too.